packages/store/addon/-private/system/store/finders.js

```diff
@@ -169,20 +168,21 @@ function ensureRelationshipIsSetToParent(payload, parentInternalModel, store, pa
       typeof relationshipData !== 'undefined' &&
       !relationshipDataPointsToParent(relationshipData, parentInternalModel)
     ) {
-      let quotedType = Ember.inspect(type);
-      let quotedInverse = Ember.inspect(inverseKey);
-      let expected = Ember.inspect({
+      let inspect = function inspect(thing) {
+        return `'${JSON.stringify(thing)}'`;
+      };
```

Comment:
FWIW, `Ember.inspect` does a bit more than `JSON.stringify` (deals with cycles, symbols, function properties, etc). It may not matter in this case (I can't tell what the actual type of `thing` is expected to be).

Reply:
nothing with a cycle :)